FRFuel.cs

```diff
@@ -88,350 +58,156 @@ public FRFuel()
-            }
-#endif
-
-            var refuelRateString = Config.Get("RefuelRate", "1");
-            if (float.TryParse(refuelRateString, out float tmpRefuelRate))
-            {
-                refuelRate = tmpRefuelRate;
-            }
-#if DEBUG
-            else
-            {
-                Debug.WriteLine("Invalid RefuelRate value. Make sure it is a valid float value, e.g. 1.2");
-            }
-#endif
+                foreach (var station in GasStations.GetInstance())
+                {
 
-            // if a valid key is set in the config file, set the control.
-            if (int.TryParse(Config.Get("EngineToggleKey", "86"), out int tmpControl))
-            {
-                engineToggleControl = (Control)tmpControl;
+                    var blip = World.CreateBlip(station.Coordinates);
+                    blip.Sprite = BlipSprite.JerryCan;
+                    blip.Color = BlipColor.White;
+                    blip.Scale = .75f;
+                    blip.IsShortRange = true;
+                    blip.Name = "Gas Station";
+                }
             }
 
-            fuelBarNormalHexColor = Config.Get("FuelBarNormalColor", fuelBarNormalHexColor).Replace("\"", "").Replace("#", "");
-            // normal color
-            int r = MathUtil.Clamp(int.Parse(fuelBarNormalHexColor.Substring(0, 2), System.Globalization.NumberStyles.HexNumber), 0, 255);
-            int g = MathUtil.Clamp(int.Parse(fuelBarNormalHexColor.Substring(2, 2), System.Globalization.NumberStyles.HexNumber), 0, 255);
-            int b = MathUtil.Clamp(int.Parse(fuelBarNormalHexColor.Substring(4, 2), System.Globalization.NumberStyles.HexNumber), 0, 255);
-
-            fuelBarWarningHexColor = Config.Get("FuelBarWarningColor", fuelBarNormalHexColor).Replace("\"", "").Replace("#", "");
-            // warning color
-            int wR = MathUtil.Clamp(int.Parse(fuelBarWarningHexColor.Substring(0, 2), System.Globalization.NumberStyles.HexNumber), 0, 255);
-            int wG = MathUtil.Clamp(int.Parse(fuelBarWarningHexColor.Substring(2, 2), System.Globalization.NumberStyles.HexNumber), 0, 255);
-            int wB = MathUtil.Clamp(int.Parse(fuelBarWarningHexColor.Substring(4, 2), System.Globalization.NumberStyles.HexNumber), 0, 255);
-            hud.UpdateBarColors(r, g, b, wR, wG, wB);
-
-#if DEBUG
-            Debug.WriteLine($"CreatePickups: {Config.Get("CreatePickups", "true")}");
-            Debug.WriteLine($"ShowHud: {Config.Get("ShowHud", "true")}");
-            Debug.WriteLine($"EngineToggleKey: {Config.Get("EngineToggleKey", "86")}");
-            Debug.WriteLine($"FuelConsumptionRate: {Config.Get("FuelConsumptionRate", "1")}");
-#endif
+            if (Config.GetInstance().CreatePickups)
+                SpawnJerryCansPeriodically(); //dont await
         }
 
-        /// <summary>
-        /// Creates blips for gas stations
-        /// </summary>
-        public void CreateBlips()
+        static async Task SpawnJerryCansPeriodically()
```

Comment:
Current behavior of jerry cans spawn is "battle-tested" and far more optimal then this implementation as we don't spawn more pickups than we need thus making room for other resources to add their own.